### PR TITLE
Don't store redundant columns in PropertyGraph Dataframes

### LINKED ISSUE #2400: [FEA]  Do not store redundant columns in `_edge_prop_dataframe ` and `_vertex_prop_dataframe `
**Describe the solution you'd like and any additional context**

When we add data to Property graph via `add_vertex_data` and `add_edge_data` we store the `vertex_columns` along with the internal `_SRC_` and `_DST_` . We should drop those to save on memory. 

```python
from cugraph.experimental import PropertyGraph
import numpy as np
import cudf

pg = PropertyGraph()
df = cudf.DataFrame({'input_src':np.ones(100_000, dtype=np.int32)*2, 'input_dst':np.arange(100_000), 'edge_w':np.zeros(100_000)})
pg.add_edge_data(df,vertex_col_names=['input_src','input_dst'])
pg._edge_prop_dataframe

pg._edge_prop_dataframe.memory_usage()
``` 

```python


_SRC_ | _DST_ | _EDGE_ID_ | _TYPE_ | input_src | input_dst | edge_w
-- | -- | -- | -- | -- | -- | --
2 | 0 | 0 | <NA> | 2 | 0 | 0.0
2 | 1 | 1 | <NA> | 2 | 1 | 0.0
2 | 2 | 2 | <NA> | 2 | 2 | 0.0
2 | 3 | 3 | <NA> | 2 | 3 | 0.0
2 | 4 | 4 | <NA> | 2 | 4 | 0.0
... | ... | ... | ... | ... | ... | ...
2 | 99995 | 99995 | <NA> | 2 | 99995 | 0.0
2 | 99996 | 99996 | <NA> | 2 | 99996 | 0.0
2 | 99997 | 99997 | <NA> | 2 | 99997 | 0.0
2 | 99998 | 99998 | <NA> | 2 | 99998 | 0.0
2 | 99999 | 99999 | <NA> | 2 | 99999 | 0.0

```

In this example the `input_src` and `input_dst` is redundant and we can save `30%` of memory by dropping them.

### PR DESCRIPTION
**Pull request by eriknw**
The main purpose of this is to reduce memory usage.

Closes #2400

I still need to update MG tests.

I'll also remove the in-code assertions, since they won't always be True, because a column name could have previously been used as a property. Nevertheless, seeing these assertions pass should give us warm-fuzzies :)

### CODE REVIEW

**Comment by eriknw** [comment]
One thing I don't like about the current state of this PR is that it makes the user use our column names for source, dest, etc. in the query string.

What if `"_SRC_ == 4"` was instead specified as e.g. `"{src} == 4"` (or `f"{{src}} == 4"` if the user was using an f-string).  Any other syntax or spelling we should consider?

**Review by rlratzel — APPROVED** [review]
Looks good. There might be places in here that will be affected by the outcome of how we want to handle the "internal" column names in output and tests, but no need to change anything now.

**Review by alexbarghi-nv — APPROVED** [review]

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Have requested changes to drop inpace for cudf columns.  LGTM otherwise.

File: python/cugraph/cugraph/structure/property_graph.py [hunk 1]
```diff
@@ -329,7 +329,9 @@ def add_vertex_data(self,
             # remove the ones to keep
             column_names_to_drop.difference_update(property_columns +
                                                    default_vertex_columns)
-            tmp_df = tmp_df.drop(labels=column_names_to_drop, axis=1)
+        else:
+            column_names_to_drop = {vertex_col_name}
+        tmp_df = tmp_df.drop(labels=column_names_to_drop, axis=1)
```
[the remark this answers is not in the document]
Comment (eriknw) [reply python/cugraph/cugraph/structure/property_graph.py:334]: Fair question.  I generally don't like to use `inplace=True`.   It often makes code harder to debug, and the performance or memory usage benefit is usually negligible.  Many pandas devs have been saying "don't use inplace" and considering deprecating it for years: https://github.com/pandas-dev/pandas/issues/16529

I don't know about `cudf`.  Does using `inplace=True` have tangible benefits when using `cudf`?

[the remark this answers is not in the document]
Comment (VibhuJawa) [reply python/cugraph/cugraph/structure/property_graph.py:334]: At least, this used to be the case where using `inplace=True`   had tangible  benefits in terms of memory usage.  @galipremsagar  , Would you know if that still holds true ?

[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cugraph/cugraph/structure/property_graph.py:334]: Using `inplace=True` will avoid making a new copy and operates on the same data frame itself. So I expect it to be a bit performant in this case.

**Review by VibhuJawa — APPROVED** [review]
LGTM. Thanks Erik

**Comment by eriknw** [comment]
Test failures are now the same as seen in other PRs:
```
    from cupyx.scipy.sparse.coo import coo_matrix as cp_coo_matrix
E   ModuleNotFoundError: No module named 'cupyx.scipy.sparse.coo'
```

**Review by BradReesWork — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**